src/braille/internal/braille.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef MU_BRAILLE_BRAILLE_H
+#define MU_BRAILLE_BRAILLE_H
+
+#include <QIODevice>
+
+#include "engraving/types/types.h"
+#include "engraving/libmscore/types.h"
+
+namespace mu::engraving {
+class Arpeggio;
+class Articulation;
+class BarLine;
+class Breath;
+class Chord;
+class ChordRest;
+class Clef;
+class DurationElement;
+class Dynamic;
+class EngravingItem;
+class Fermata;
+class Fingering;
+class Hairpin;
+class Jump;
+class KeySig;
+class Lyrics;
+class Marker;
+class Measure;
+class MeasureRepeat;
+class MMRest;
+class Note;
+class Rest;
+class Score;
+class Slur;
+class TempoText;
+class TimeSig;
+class Tuplet;
+class Volta;
+
+class BrailleEngravingItems
+{
+public:
+    BrailleEngravingItems();
+    ~BrailleEngravingItems();
+
+    void clear();
+
+    void join(BrailleEngravingItems*, bool newline = true, bool del = true);
+    void join(std::vector<BrailleEngravingItems*>, bool newline = true, bool del = true);
+
+    QString brailleStr();
+    std::vector<std::pair<EngravingItem*, std::pair<int, int> > >* items();
+
+    void setBrailleStr(QString str);
+    void addPrefixStr(QString str);
+
+    void addEngravingItem(EngravingItem*, QString braille);
+    void addLyricsItem(Lyrics*);
+
+    bool isEmpty() { return braille_str.isEmpty(); }
+    EngravingItem* getEngravingItem(int pos);
+    std::pair<int, int> getBraillePos(EngravingItem* e);
+
+    void log();
+private:
+    QString braille_str;
+    std::vector<std::pair<EngravingItem*, std::pair<int, int> > > _items;
```

Comment:
please use `m_camelCase` for members https://github.com/musescore/MuseScore/wiki/CodeGuidelines